ICSharpCode.Decompiler/IL/Transforms/SwitchOnStringTransform.cs

```diff
@@ -702,6 +702,8 @@ bool HasLabel(SwitchSection section)
 			{
 				if (section == defaultSection)
 					continue;
+				if (nullValueCaseBlock != null && section.Body.MatchBranch(nullValueCaseBlock))
```

Comment:
This if seems to say "sections that join up with `case null` cannot be used as the default section".
Why would that make any sense?

Reply:
At that stage of the transform the first section is usually the default section (that section's label contains all "unused" values). We want to exclude "case null" because it is treated specially in the transform. mcs 5 (unlike mcs 2 or any MS compiler) seems to generate an entry for case null in the jump-table, which causes "FixCasesWithoutValue" to fail.